components/ai_chat/core/ai_chat_helper.h

```diff
@@ -0,0 +1,158 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_AI_CHAT_HELPER_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_AI_CHAT_HELPER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/ref_counted.h"
+#include "base/observer_list.h"
+#include "brave/components/ai_chat/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/engine/engine_consumer.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "services/data_decoder/public/cpp/data_decoder.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+
+class PrefService;
+
+namespace ai_chat {
+class AIChatMetrics;
+
+class AIChatHelperDelegate {
+ public:
+  virtual raw_ptr<PrefService> GetPrefsService() const = 0;
+  virtual raw_ptr<AIChatMetrics> GetChatMetrics() const = 0;
+  virtual scoped_refptr<network::SharedURLLoaderFactory> GetURLLoaderFactory()
+      const = 0;
+  virtual GURL GetPageURL() const = 0;
+  virtual void GetPageContent(
+      base::OnceCallback<void(std::string, bool is_video)> callback) const = 0;
+  virtual bool HasPrimaryMainFrame() const = 0;
+  virtual bool IsDocumentOnLoadCompletedInPrimaryMainFrame() const = 0;
+};
+
+class AIChatHelper {
+ public:
+  class Observer : public base::CheckedObserver {
+   public:
+    ~Observer() override {}
+
+    virtual void OnHistoryUpdate() {}
+    virtual void OnAPIRequestInProgress(bool in_progress) {}
+    virtual void OnAPIResponseError(ai_chat::mojom::APIError error) {}
+    virtual void OnSuggestedQuestionsChanged(
+        std::vector<std::string> questions,
+        bool has_generated,
+        ai_chat::mojom::AutoGenerateQuestionsPref auto_generate) {}
+    virtual void OnFaviconImageDataChanged() {}
+    virtual void OnPageHasContent() {}
+  };
+
+  explicit AIChatHelper(AIChatHelperDelegate* delegate);
```

Comment:
Please add comments explaining what this class does

Reply:
also `tab_helper` is a chromium naming convention, but I'm not sure that `AIChatHelper` is the right name for this. It seems like it has a lot of responsibilities and I'm not sure it makes sense as a single class.

Reply:
I'm not opposed to this as an intermediate refactoring with a follow-up to clean this up, but it definitely needs some comments

Reply:
Renamed to `AIChatDriver` and refactored the code to get rid of the delegate and follow the `Driver` style.
So TabHelper now implements the driver and iOS will do the same.

Reply:
Didn't see this conversation when I just posted my other messages about this so adding here that I think naming this `Conversation` and using a Delegate is the correct approach. If not now, then very soon as follow-up or once we land support for standalone `Conversation`s which are not tied to a Tab / Content. The plan for that was to use a delegate to fetch an associated Tab for a Conversation. There would only be an associated Tab if 1) the conversation was started from the sidepanel of a Tab and 2) the conversation was not loaded from persistant storage.